packages/pds/src/actor-store/index.ts

```diff
@@ -0,0 +1,220 @@
+  }
+
+  async keypair(did: string): Promise<Keypair> {
+    const got = await this.keyCache.fetch(did)
+    if (!got) {
+      throw new InvalidRequestError('Keypair not found', 'NotFound')
+    }
+    return got
+  }
+
+  async db(did: string): Promise<ActorDb> {
+    const got = await this.dbCache.fetch(did)
+    if (!got) {
+      throw new InvalidRequestError('Repo not found', 'NotFound')
+    }
+    return got
+  }
+
+  async reader(did: string) {
+    const [db, keypair] = await Promise.all([this.db(did), this.keypair(did)])
+    return createActorReader(did, db, keypair, this.resources)
+  }
+
+  async read<T>(did: string, fn: ActorStoreReadFn<T>) {
+    const reader = await this.reader(did)
+    return fn(reader)
+  }
+
+  async transact<T>(did: string, fn: ActorStoreTransactFn<T>) {
+    const [db, keypair] = await Promise.all([this.db(did), this.keypair(did)])
+    return db.transaction((dbTxn) => {
+      const store = createActorTransactor(did, dbTxn, keypair, this.resources)
+      return fn(store)
+    })
+  }
+
+  async create<T>(
+    did: string,
+    keypair: ExportableKeypair,
+    fn: ActorStoreTransactFn<T>,
+  ) {
+    const { subdir, dbLocation, keyLocation } = await this.getLocation(did)
+    // ensure subdir exists
+    await mkdir(subdir, { recursive: true })
+    const exists = await fileExists(dbLocation)
+    if (exists) {
+      throw new InvalidRequestError('Repo already exists', 'AlreadyExists')
+    }
+    const db: ActorDb = getDb(dbLocation)
+    const migrator = getMigrator(db)
+    const privKey = await keypair.export()
+    await Promise.all([
+      await migrator.migrateToLatestOrThrow(),
+      await fs.writeFile(keyLocation, privKey),
+    ])
+
+    const result = await db.transaction((dbTxn) => {
+      const store = createActorTransactor(did, dbTxn, keypair, this.resources)
+      return fn(store)
+    })
+    this.dbCache.set(did, db)
+    return result
+  }
+
+  async destroy(did: string) {
+    const blobstore = this.resources.blobstore(did)
+    if (blobstore instanceof DiskBlobStore) {
+      await blobstore.deleteAll()
+    } else {
+      const db = await this.db(did)
+      const blobRows = await db.db.selectFrom('blob').select('cid').execute()
+      const cids = blobRows.map((row) => CID.parse(row.cid))
+      await Promise.allSettled(cids.map((cid) => blobstore.delete(cid)))
+    }
+
+    const got = this.dbCache.get(did)
+    this.dbCache.delete(did)
+    if (got) {
+      await got.close()
+    }
+
+    const { dbLocation, keyLocation } = await this.getLocation(did)
+    await rmIfExists(dbLocation)
+    await rmIfExists(`${dbLocation}-wal`)
+    await rmIfExists(`${dbLocation}-shm`)
+    await rmIfExists(keyLocation)
+  }
+
+  async close() {
+    const promises: Promise<void>[] = []
+    for (const key of this.dbCache.keys()) {
+      const got = this.dbCache.get(key)
+      this.dbCache.delete(key)
+      if (got) {
+        promises.push(got.close())
+      }
```

Comment:
I think that the `delete()` also attempts to dispose of the db.  Maybe the close is idempotent and a non-issue.  If it is an issue, I think the dispose function takes a `reason` parameter, and you could explicitly skip the disposal on deletes.

Reply:
the close should be idempotent

I haven't noticed any issues from this in tests. One downside of relying on the cache to handle the dispose is that the `.delete()` function is synchronous & I don't see a way to track when it has finished disposing of the item

Reply:
If we end-up needing to address that, we could keep a p-queue for disposing dbs and add to it during lrucache's `dispose()`.